app/helpers/spree/admin/orders_helper_decorator.rb

```diff
@@ -5,28 +5,38 @@ def order_links(order)
         @order ||= order
         links = []
         links << edit_order_link unless action_name == "edit"
-        if @order.complete?
-          links << resend_confirmation_link
-          if @order.distributor.can_invoice?
-            links << send_invoice_link_with_url
-          else
-            links << send_invoice_link_without_url
-          end
-          links << print_invoice_link
-          if Spree::Config.enable_receipt_printing?
-            links << print_ticket_link
-            links << select_ticket_printer_link
-          end
-        end
-        if @order.ready_to_ship?
-          links << ship_order_link
-        end
+        links << complete_order_links if @order.complete?
+        links << ship_order_link if @order.ready_to_ship?
         links << cancel_order_link if @order.can_cancel?
         links
       end
 
       private
 
+      def complete_order_links
+        complete_order_links = []
+        complete_order_links << resend_confirmation_link
+        complete_order_links << invoice_links
+        complete_order_links << ticket_links
+        complete_order_links
+      end
+
+      def invoice_links
+        invoice_links = []
+        invoice_links << if @order.distributor.can_invoice?
```

Comment:
If you introduce a new `send_invoice_link` method you can write:
```rb
  [send_invoice_link, print_invoice_link]
```

Reply:
nice, thanks! new commit.